domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -922,6 +933,54 @@ public void save(AsyncResult<SQLConnection> sqlConnection, String table, String
     }
   }
 
+  /**
+   * Save entity in table and return the updated entity.
+   *
+   * @param sqlConnection connection (for example with transaction)
+   * @param table where to insert the entity record
+   * @param id  the value for the id field (primary key)
+   * @param entity  the record to insert, a POJO
+   * @param replyHandler  where to report success status and the entity after applying any database INSERT triggers
+   */
+  private <T> void saveAndReturnUpdatedEntity(AsyncResult<SQLConnection> sqlConnection, String table, String id, T entity,
+      Handler<AsyncResult<T>> replyHandler) {
```

Comment:
Can you handle `id == null` by generating a random UUID and adjust the javadoc accordingly?